kv/dist_sender.go

```diff
@@ -600,10 +618,48 @@ func (ds *DistSender) sendAttempt(trace *tracer.Trace, args proto.Request, desc
 // This may temporarily adjust the request headers, so the proto.Call
 // must not be used concurrently until Send has returned.
 func (ds *DistSender) Send(ctx context.Context, call proto.Call) {
-	args := call.Args
+	// TODO(tschottdorf): provisional code that wraps everything in a one-
```

Comment:
there's a fair bit of ugliness below - what's the plan with that? are you looking to merge this as-is and continue work in future PRs or is this a WIP?